js/messaging.js

```diff
@@ -370,15 +370,47 @@ function deleteStyleSafe({id, notify = true} = {}) {
 
 function download(url) {
   return new Promise((resolve, reject) => {
+    url = new URL(url);
+    const TIMEOUT = 10000;
+    const options = {
+      method: url.search ? 'POST' : 'GET',
+      body: url.search ? url.search.slice(1) : null,
+      headers: {
+        'Content-type': 'application/x-www-form-urlencoded'
+      }
+    };
+    if (url.protocol === 'file:' && navigator.userAgent.includes('Firefox')) {
```

Comment:
navigator.userAgent may be spoofed in FF before 57 so switch to using `FIREFOX` variable which is correctly determined in `master` branch